src/transports/http.c

```diff
@@ -367,11 +367,15 @@ static int on_headers_complete(http_parser *parser)
 								  allowed_auth_types,
 								  t->owner->cred_acquire_payload);
 
-				if (error == GIT_PASSTHROUGH) {
-					no_callback = 1;
-				} else if (error < 0) {
+				/* treat GIT_PASSTHROUGH as if callback isn't set */
+				if (error == GIT_PASSTHROUGH)
+					error = 1;
```

Comment:
This makes me wonder whether we should just document that `GIT_PASSTHROUGH` is treated in a special way instead of accepting positive return values. Especially with regards to consistency with other user-provided callback functions, where we say that we try to pass back positive error values unchanged.